spec.md

```diff
@@ -230,6 +245,49 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Public Modules
+
+A public module is accessible by `Circuit`s other than the `Circuit` it is
+defined in. Unless otherwise specified, all modules inside of a `Circuit` are
+private and cannot be accessed by any other `Circuit`.
```

Comment:
A design point here is should the main module be default-public and should the main module be required to use the public keyword? I think it's reasonable to say that the main module is always public and does not use the public keyword. (This matches the current semantic for how the main module has some special handling around port removal right now, but this semantic could be changed.)